ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py

```diff
@@ -116,6 +116,23 @@ def test_numpy_ndarray_property_T(
     )
 
 
+@given(
+    dtype_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid", prune_function=False),
+        ret_shape=True,
+    )
+)
+def test_numpy_ndarray_property_flat(dtype_x):
+    dtype, data, shape = dtype_x
+
+    x = ndarray(shape, dtype[0])
+    x.ivy_array = data[0]
+
+    flat_ivy = x.flat
+    flat_generated = data[0].reshape(-1)
+    ivy.utils.assertions.check_equal(flat_ivy, flat_generated, as_array=True)
```

Comment:
The way you're testing this is wrong, as `check_equal` are only used for non tensor inputs, i.e. int, float, list, tuple...

You should be doing this:
```python
flat_ivy = x.flat
flat_generated = data[0].flatten()
flat_ivy = helpers.flatten_and_to_np(ret=flat_ivy)
flat_generated = helpers.flatten_and_to_np(ret=flat_generated)

assert np.allclose(flat_ivy, flat_generated)
```

Reply:
Now that I coded this out, I realized this is wrong and defeats the entire purpose of testing for `flatten` :sweat_smile: Please find another helper/way to convert them into numpy and then use `np.allclose` without using `flatten_and_to_np`

Reply:
> 
@xoiga123 since i have converted the arrays into a flat array so we can check it using check_equal noh?(they arent tensor any more)...moreover, we need to return a 1d iterator over the array elements noh?